src/Marten/Linq/QueryHandlers/QueryHandlerFactory.cs

```diff
@@ -157,7 +162,7 @@ private IQueryHandler<T> tryFindSingleQuery<T>(QueryModel model, IIncludeJoin[]
 
             var setters = findSetters(queryType, expression);
 
-            var model = MartenQueryParser.Flyweight.GetParsedQuery(invocation);
+            var model = MartenQueryParser.ForCachedQuery.GetParsedQuery(invocation);
```

Comment:
I had to add another instance- for a compiled query. Flyweight is still there: https://github.com/JasperFx/marten/pull/323/files#diff-11bbc3e3b8d500d2e3747762350d973dL11